### ISSUE TITLE
Momentum Wifi Camera Solution

### ISSUE THREAD

**Issue by Jborn4christ** [KhaosT/homebridge-camera-ffmpeg#80]
So Walmart and Amazon sell a nice little wifi camera, but for months I have not been able to configure it properly for Homebridge. After messing around last night, and using some tips from iSpyConnect as well as a few other places, I was able to get them working!! I'm including the code from my config.json for anyone else, as well as the link to the cameras, as they are very affordable and robust. After relying on all of you to help me get my stuff running efficiently, it feels good to contribute even on a small scale. I currently have 4 of them working flawlessly in homebridge with still image previews.

Cameras: https://www.amazon.com/Momentum-WiFi-Home-Security-Camera/dp/B018F521QC

Code:
{
		"platform": "Camera-ffmpeg",
		"cameras": [{
			"name": "any name",
			"videoConfig": {
				"source": "-re -i rtsp://admin:admin@yourcameraipaddress:8001/H264/sub",
				"stillImageSource": "-i http://admin:admin@yourcameraipaddress/snapshot.cgi?user=admin&pwd=admin",
				"maxStreams": 2,
				"maxWidth": 1280,
				"maxHeight": 720,
				"maxFPS": 30
			}
		}]
	}

**Comment by thechrisreese** [comment]
@Jborn4christ I am moving my comment here. 

other than IP address do I need to change anything else? I have this set up in config.json but it doesn't load the platform.

**Comment by thechrisreese** [comment]
Ok I have it to the point of pairing in Home app but it just spins without completing.

**Comment by thechrisreese** [comment]
@jborn4christ I messed with it this weekend and got it all running. You rock for getting this running with this camera. Cheap and easy. Now to buy 5 more......

**Comment by thechrisreese** [comment]
Question: When do the image previews Update for you? 

Also is there a way to get motion alerts that you know of?

**Comment by thechrisreese** [comment]
Ok I have tried to get the drive uploads to work to no success. 

I cannot tell if the stillImageSource is actually working on this or not. I do not get any updated Still Images unless I open the stream.

**Comment by NorthernMan54** [comment]
@Jborn4christ we have started recording working configurations in the wiki.  You may want to post this there.

@thechrisreese if you only get the still from opening the stream, then it looks like your still image source is not working.   Also, in the latest build, it will log the still image command so you can see when it calls for stills.

**Comment by thechrisreese** [comment]
@NorthernMan54 I see it logging the call for stills (no time readout but I can guesstimate) and was hoping there was a local file versus the Drive. I was thinking Drive wasn't working but maybe its the stillImageSource that isnt working. 

What makes it call for a new still? Motion by the device or Camera-ffmpeg refresh?

**Comment by NorthernMan54** [comment]
@thechrisreese Stills are controlled by the Home app, and when ever it decides to do a refresh.   I find that on WIFI, it refreshes every 10 seconds and on mobile data every 60 seconds when you are in the room with camera.

To debug this, you can take the command line from your log file, and just run it to make sure it works.  I added ffmpeg to the start, and changed the last dash '-' to a filename.  The filename should be a picture.

ffmpeg -i http://alice:8080/shot.jpg -t 1 -s 1920x1080 -f image2 test.jpg

**Comment by thechrisreese** [comment]
So changed to "ffmpeg http://admin:admin@10.0.1.92/snapshot.cgi?user=admin&pwd=admin -t 1 -s 480x270 -f image2 test.jpg" 

And get this:
- bash: -t: command not found
Thoughts on this?

**Comment by NorthernMan54** [comment]
On the linux command line, the % character is reserved so you need to backslash when you test on the command line.  This is not needed in the config.json i.e.

ffmpeg http://admin:admin@10.0.1.92/snapshot.cgi?user=admin\&pwd=admin -t 1 -s 480x270 -f image2 test.jpg

PS I think your missing the -i at the start, but ymmv

**Comment by Jborn4christ** [comment]
@thechrisreese Let me know if you figure out the still image handling. I can't get mine to refresh periodically, but it will keep the last frame of the video I view.

**Comment by Jborn4christ** [comment]
@NorthernMan54 Want to wait until the bugs are all figured out (by those far smarter than I) before posting there. I'm not a coder, just a guy that was hell bent on figuring out a way to make these cameras work.

**Comment by thechrisreese** [comment]
@Jborn4christ the key is that stillImageSource doesnt work. Every time it is called nothing is retrieved so we would have to find the real link. Haven't had success so far.

**Comment by thechrisreese** [comment]
@Jborn4christ Question - do you have motion detection turned on for your Momentum Cam's?

**Comment by Jborn4christ** [comment]
@thechrisreese any luck on still image or motion detector?

**Comment by thechrisreese** [comment]
@Jborn4christ nope never got it to work. Even bought a motion detector to try and make it work. No luck.

**Comment by Jborn4christ** [comment]
Bummer...

**Comment by Grapplerone** [comment]
I know this is an old post but I have 3 of these exact cams. I've tried to get these working in home bridge using the Homebridge Camera FFMPEG plug in and cannot get them to work in HomeKit. I keep getting this response in Hombridge with no video or snapshot appearing in HomeKit. I have tried using my internal ip:8001, external ip (is static) forwarded to 8001, just my internal IP. Also for admin:admin, I used my Momentum Login which is my email & Password. I noticed under the frame capture it is requested TWICE which is odd. I even tried it be leaving those as admin:admin with the same results. Any idea why it worked for you but I cannot get them working? I purchased these a long time back on Amazon, they show on firmware 1.1.7 and up to date in the app and from a sniffer.

[11/20/2023, 1:24:49 PM] [Camera FFmpeg] [GarageCam] Starting video stream: 1280 x 720, 30 fps, 299 kbps
[11/20/2023, 1:24:52 PM] [Camera FFmpeg] [GarageCam] Stopped video stream.
[11/20/2023, 1:24:52 PM] [Camera FFmpeg] [GarageCam] Failed to fetch snapshot.
[GarageCam 662D] Snapshot request handler provided empty image buffer!
